### PR TITLE
Added ITDE manager

### LINKED ISSUE #44: Add function to start the ITDE and add its credentials to a configuration
ITDE has an API for spawning a DockerDB:

https://github.com/exasol/integration-test-docker-environment/blob/cbceeed7d4fb53872751b0dee74f7ec3dc1aa517/exasol_integration_test_docker_environment/lib/api/spawn_test_environment.py#L35

Port-Forwards won't work with DockerDB Image of the AI-Lab.

Users might want to configure the parameters:
        db_mem_size: str = "2 GiB",
        db_disk_size: str = "2 GiB",
        nameserver: Tuple[str, ...] = tuple(),
        create_certificates: bool = False,
        additional_db_parameter: Tuple[str, ...] = tuple(),  # Maybe hidden

The API returns an EnvironmentInfo object, besides other things. The EnvironmentInfo object contains IP and Ports of the DB and the DB Credentials.

The bucketfs credentials are hard-coded to r:read and w:write.

What happens, if the user runs the start ITDE again. For example, he stopped all docker containers and restart them now.

We need to check during connect if the docker-db is running and, if not, start it. When the user uses the function for initial start, we need a warning that this will create a new docker-db and that data will get lost. The user confirms this before we proceed.

Checking if the dockerdb is running:

The EnvironmentInfo contains a DatabaseInfo which contains a ContainerInfo with the name of the docker container of the DB. We store the container-name in the secret store when creating the docker-db and during connect we get the container-name from the secret store, ask docker if the container with this name exists, if not error message. Then we ask if it is running, if not we restart. 

start a docker-container from python: https://docker-py.readthedocs.io/en/stable/containers.html#docker.models.containers.Container.start

determine of a container exists: https://docker-py.readthedocs.io/en/stable/containers.html#docker.models.containers.ContainerCollection.get

found if it exists: https://docker-py.readthedocs.io/en/stable/containers.html#docker.models.containers.Container.status

Some diagrams:

ai-lab-docker-db drawio

### PR DESCRIPTION
**Pull request by ahsimb**
closes #44

### CODE REVIEW

**Comment by ahsimb** [comment]
Not sure I am doing the right thing with the port forwarding. May need some advice.

**Review by tkilias — CHANGES_REQUESTED** [review]

File: exasol/itde_manager.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    conf.save(AILabConfig.bfs_user.value, 'w')
+    conf.save(AILabConfig.bfs_password.value, 'write')
+    conf.save(AILabConfig.bfs_service.value, 'bfsdefault')
+    conf.save(AILabConfig.bfs_bucket.value, 'default')
+    conf.save(AILabConfig.db_encryption.value, 'True')
+    # The bucket-fs encryption is turned off temporarily.
+    conf.save(AILabConfig.bfs_encryption.value, 'False')
+    conf.save(AILabConfig.cert_vld.value, 'False')
+
+
```
[the remark this answers is not in the document]
Comment (tkilias) [reply exasol/itde_manager.py:74]: different ticket

File: exasol/itde_manager.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    The names of the docker container, docker volume and docker network
+    are taken from the provided secret store. If the names are not found
+    there no action is taken.
+    """
+
+    remove_container(conf)
+    remove_volume(conf)
+    remove_network(conf)
+
+    conf.remove(AILabConfig.db_host_name.value)
```
Comment (ahsimb) [inline exasol/itde_manager.py:109]: Would it be better to do this in a loop:

```
for entry in AILabConfig:
   conf.remove(entry.value)

```

Comment (tkilias) [reply exasol/itde_manager.py:109]: Probably, a good Idea, at least If there are no exceptions. And, I think, the mem_size and disk_size are exceptions at the moment. I think, that is something for another PR. We need anyway another one.

**Review by tkilias — APPROVED** [review]

**Review by Shmuma — APPROVED** [review]

**Review by Shmuma — APPROVED** [review]

**Pull request merged**